src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -3077,6 +3077,41 @@ because xbuild doesn't support framework reference assemblies.
   </CalculateProjectDependencies>
 </Target>
 
+<PropertyGroup Condition="'$(BuildingInsideVisualStudio)' == 'true'">
+  <EnableCleanAllFiles Condition="'$(EnableCleanAllFiles)' == ''">true</EnableCleanAllFiles>
+  <EnableCleanOnNuGetPackagesChange Condition="'$(CleanOnNuGetPaEnableCleanOnNuGetPackagesChangeckagesChange)' == ''">true</EnableCleanOnNuGetPackagesChange>
+  <CoreCleanDependsOn Condition="'$(EnableCleanAllFiles)' == 'true'">
+     _CleanAllFiles;
+    $(CoreCleanDependsOn)
+  </CoreCleanDependsOn>
+  <BuildDependsOn Condition="'$(EnableCleanOnNuGetPackagesChange)' == 'true'">
+    _CleanOnNuGetPackagesChange;
+    $(BuildDependsOn)
+  </BuildDependsOn>
+</PropertyGroup>
+
+<Target Name="_CleanAllFiles">
+  <ItemGroup>
+    <Clean Include="$(OutputPath)**\*.*" />
+    <Clean Include="$(IntermediateOutputPath)**\*.*" />
```

Comment:
Won't this screw up the Design-Time build? I'm reminded of 01be8ac8314211095a3f69e020f2b2fcc5a66865 / Issue #1286, in which a Rebuild ("Clean", "Build") would *break* IntelliSense, as the directories that IntelliSense was using were *removed*.

The directories that IntelliSense uses includes `$(IntermediateOutputPath)\designtime`, which *this* PR would *remove*. I don't think that is desirable.

Reply:
Aside lol: the whole reason the unit tests added in 01be8ac pass is because this new behavior only applies when `$(BuildingInsideVisualStudio)`=True, which isn't the case for the unit tests.

Consequently, the scenario/environment that 01be8ac was attempting to fix/address -- use within Visual Studio -- is precisely the environment that this PR will change, in an *untested* manner.